editor.planx.uk/src/pages/FlowEditor/lib/store/navigation.ts

```diff
@@ -97,16 +100,22 @@ export const navigationStore: StateCreator<
 
     const mostRecentSectionId = findLast(
       breadcrumbIds,
-      (breadcrumbId: string) => sectionIds.includes(breadcrumbId),
+      (breadcrumbId: string) => sectionIdsSet.has(breadcrumbId)
     );
 
+    const hasPassedFirstSection = Boolean(mostRecentSectionId);
+
+    // No sections in breadcrumbs, first section values already set in store
+    if (!hasPassedFirstSection) return;
+
     // No sections in breadcrumbs, first section values already set in store
     if (!mostRecentSectionId) return;
 
     const currentSectionTitle = sectionNodes[mostRecentSectionId].data.title;
     const currentSectionIndex = sectionIds.indexOf(mostRecentSectionId) + 1;
+    const sectionProgress = _calculateSectionProgress(currentSectionIndex);
 
-    set({ currentSectionTitle, currentSectionIndex });
+    set({ currentSectionTitle, currentSectionIndex, sectionProgress });
```

Comment:
Before `set()` was being called multiple times resulting in re-renders - now it's all batched as a single update.